scripts/gen_vimdoc.py

```diff
@@ -905,49 +906,130 @@ class FunctionDoc:
     """Parameters documentation. Key is parameter name, value is doc."""
 
     doc: List[str]
-    """Main docstring (description) for the function."""
+    """Main description for the function. Separated by paragraph."""
 
     return_: List[str]
-    """Return:, or Return (multiple):"""
+    """Return:, or Return (multiple): (@return strings)"""
 
     seealso: List[str]
-    """See also:"""
+    """See also: (@see strings)"""
 
-    # for fmt_node_as_vimhelp
-    desc_node: Element | None = None
-    brief_desc_node: Element | None = None
+    xrefs: List[str]
+    """XRefs. Currently only used to track Deprecated functions."""
 
     # for INCLUDE_C_DECL
     c_decl: str | None = None
 
+    prerelease: bool = False
+
+    def export_mpack(self) -> Dict[str, Any]:
+        """Convert a dict to be exported as mpack data."""
+        exported = self.__dict__.copy()
+        del exported['notes']
+        del exported['c_decl']
+        del exported['prerelease']
+        del exported['xrefs']
```

Comment:
Question: I have no idea what those `*.mpack` files are for (see #11296), but is it OK to include additional fields if they are backward-compatible? I tried to make the generated file identical but it might be OK to include them.

Reply:
It's for language clients (and GUIs) to programmatically validate API calls.

Reply:
Thanks. I see they are also used during the build process by `gen_eval_files.lua` to generate `GEN_EVAL_FILES`.